src/AutoRest.CSharp/Mgmt/Generation/ResourceWriter.cs

```diff
@@ -1,37 +1,57 @@
 using AutoRest.CSharp.Mgmt.Output;
-using AutoRest.CSharp.Output.Models.Types;
 using Azure.ResourceManager.Core;
 
 namespace AutoRest.CSharp.Mgmt.Generation
 {
     internal class ResourceWriter
     {
-        public void WriteResource(CodeWriter writer, BuildContext<MgmtOutputLibrary> context, Resource resource)
+        public void WriteResource(CodeWriter writer, Resource resource)
         {
-            using (writer.Namespace(resource.Type.Namespace))
+            var cs = resource.Type;
+
+            using (writer.Namespace(cs.Namespace))
             {
-                var resourceOperation = context.Library.GetResourceOperation(resource.OperationGroup);
                 writer.WriteXmlDocumentationSummary(resource.Description);
-                using (writer.Scope($"{resource.Declaration.Accessibility} class {resource.Type.Name} : {resourceOperation.Type}"))
+
+                using (writer.Scope($"{resource.Declaration.Accessibility} class {cs.Name} : {cs.Name}Operations"))
                 {
-                    var resourceData = context.Library.GetResourceData(resource.OperationGroup);
-                    WriteConstructors(writer, resource, resourceData);
-                }
-            }
-        }
+                    // internal constructor
+                    writer.WriteXmlDocumentationSummary($"Initializes a new instance of the <see cref = \"{cs.Name}\"/> class.");
+                    writer.WriteXmlDocumentationParameter("options", "The client parameters to use in these operations.");
+                    writer.WriteXmlDocumentationParameter("resource", "The resource that is the target of operations.");
+                    using (writer.Scope($"internal {cs.Name}({typeof(ResourceOperationsBase)} options, {resource.ResourceDataObject.Type} resource) : base(options, resource.Id)"))
+                    {
+                        writer.LineRaw("Data = resource;");
+                    }
 
-        private void WriteConstructors(CodeWriter writer, Resource resource, ResourceData resourceData)
-        {
-            writer.WriteXmlDocumentationSummary($"Initializes a new instance of the <see cref=\"{resource.Type.Name}\"/> class.");
-            writer.WriteXmlDocumentationParameter("options", "The client parameters to use in these operations.");
-            writer.WriteXmlDocumentationParameter("resource", "The resource that is the target of operations.");
-            using (writer.Scope($"internal {resource.Type.Name}({typeof(ResourceOperationsBase)} options, {resourceData.Type} resource) : base()"))
-            {
+                    // write Data
+                    writer.Line();
+                    var x = resource.ResourceDataObject.GetType();
+                    writer.WriteXmlDocumentationSummary($"Gets or sets the {resource.ResourceDataObject.Type.Name}.");
+                    writer.LineRaw("public " + resource.ResourceDataObject.Type.Name + " Data {get; private set;}");
```

Comment:
Don't concatenate strings (`resource.ResourceDataObject.Type.Name + Data`) here. We should use the ResourceData Type.

You can call [GetResourceData(operationGroup)](https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs#L77) method to get the type.